docs/RationaleHW.md

```diff
@@ -142,6 +142,42 @@ notable differences: for example:
    on the other hand, do support both of these.  Zero width ports are omitted (printed as
    comments) when generating verilog.
 
+### GlobalRef
+The GlobalRef operation (`hw.globalRef`) can be used to identify the unique
+ instance path of an operation globally.
+`hw.globalRef` can be used to attach nonlocal annotations in FIRRTL dialect
+and also for metadata emission.
+`hw.globalRef` defines a symbol and contains a list of module local
+`hw.innerNameRef` symbols to define the instance path.
+ For example, in the following example, `@glbl_B_M1` specifies instance
+ "h1" in module `@A`, followed by instance
+ "M1" in module `@B`.
+
+`hw.globalRef` can define a unique instance path, and each element along the way
+ carries an attribute `circt.globalRef`, pointing to the global op. 
+ Thus instances participating in nonlocal paths are readily apparent.
```

Comment:
This property should be verified.  A globalRef needs a verifier that checks that each thing it references points back to it.

Reply:
Added a verifier, except for ports. Not sure how to deal with inner symbols for FIRRTL vs HW.